### PR TITLE
Fixes #11045. Assigning a proc to const and invoking.

### LINKED ISSUE #11045: `const` alias to compile time function fails.
I need some help with a compiler bug I'm trying to resolve. Currently the following fails with an error about trying to run a compile time proc at runtime:
```
proc foo(): int {.compileTime.} = 100

const Foo = foo

echo Foo()
```
I dug through the compiler and the problem is that `const` substitution happens after compile time calls are resolved. 

This works:
```
const Foo = foo()
echo Foo
```
and so does this:
```
echo foo()
```
I spent another few hours trying to come up with a nice solution but failed. I feel like the best solution at this point is to make aliasing a `const` to a compile time `proc` an error. Any help is appreciated.

### PR DESCRIPTION
**Pull request by deech**

### CODE REVIEW

**Comment by deech** [comment]
These test failures seem spurious. I'm getting them even without these changes.

**Comment by deech** [comment]
@Araq Any feedback?

File: compiler/semexprs.nim [hunk 1]
```diff
@@ -707,19 +707,17 @@ proc evalAtCompileTime(c: PContext, n: PNode): PNode =
           break maybeLabelAsStatic
       n.typ = newTypeWithSons(c, tyStatic, @[n.typ])
       n.typ.flags.incl tfUnresolved
-
```
Comment (Araq) [inline compiler/semexprs.nim]: Don't rewrite logic to suit your style, now the fix is unreadable. Is the fix as simple as ``callee.kind notin {skProc, skFunc, skConverter, skConst}``? Who knows.

Comment (deech) [reply compiler/semexprs.nim]: Ok I'll revert it and just add the relevant bit back. Just thought I'd clean up since I was there.

**Comment by deech** [comment]
@Araq The refactoring has now been removed per your comment.

**Pull request merged**